bundles/org.eclipse.text/src/org/eclipse/jface/text/FindReplaceDocumentAdapter.java

```diff
@@ -199,7 +196,7 @@ private IRegion findReplace(final FindReplaceOperationCode operationCode, int st
 					replaceText= interpretReplaceEscapes(replaceText, prevMatch);
 					Matcher replaceTextMatcher= pattern.matcher(prevMatch);
 					replaceText= replaceTextMatcher.replaceFirst(replaceText);
-				} catch (IndexOutOfBoundsException ex) {
+				} catch (IndexOutOfBoundsException | IllegalArgumentException ex) {
```

Comment:
This is necessary to fulfill the intention of `interpretReplaceEscapes()` to leave an unfinished character escape at the end of the pattern as is (such as in the pattern `demo\`) in order to produce a proper error message (which is delivered through an `IllegalArgumentException` from `Matcher#replaceFirst()`).

Reply:
Indeed, I have added a comment to the code explaining why these exceptions are caught.